primo/data_parser/metric_data.py

```diff
@@ -0,0 +1,602 @@
+
+# User-defined libs
+from primo.utils.config_utils import UserPriorities
+from primo.data_parser.default_data import (
+    SUPP_IMPACT_METRICS,
+    SUPP_EFF_METRICS,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable = protected-access
+# pylint: disable = too-many-arguments
+# pylint: disable = too-many-instance-attributes
+class Metric:
+    """
+    Dataclass for creating metric objects
+    """
+
+    def __init__(
+        self,
+        name,
+        value,
+        min_value=0,
+        max_value=100,
+        full_name=None,
+    ) -> None:
+        """
+        Returns a `Metric` object
+
+        Parameters
+        ----------
+        name : str
+            Short name of the metric.
+            Must be a valid python variable name
+
+        value : float
+            Weight associated with the metric
+
+        min_value : float, default = 0
+            Lower bound on metric value
+
+        max_value : float, default = 100
+            Upper bound on metric value
+
+        full_name : str, default = None
+            Elaborate name of the metric
+
+        Raises
+        ------
+        ValueError
+            If an invalid python variable name is provided,
+            If `value` lies outside the bounds,
+            If an attempt is made to overwrite `name`
+        """
+
+        self.name = name
+        self.full_name = name if full_name is None else full_name
+        self.is_submetric = False
+        self.min_value = min_value
+        self.max_value = max_value
+        self.value = value
+        # Name of the column that contains the data
+        # needed for calculating priority score
+        self.data_col_name = None
+        # Name of the column that contains the priority score
+        self._score_col_name = None
+
+    def __str__(self) -> str:
+        """Format for printing the object"""
+        return (
+            f"Metric name: {self.full_name}, Metric value: {self.value} \n"
+            f"    Admissible range: [{self.min_value}, {self.max_value}]"
+        )
+
+    @property
+    def name(self):
+        """Getter for attribute `name`"""
+        return self._name
+
+    @name.setter
+    def name(self, val):
+        """Setter for attribute `name`"""
+        if hasattr(self, "_name"):
+            raise ValueError(
+                "Metric's key name cannot be modified " "after it is defined."
+            )
+
+        if not val.isidentifier():
+            raise ValueError(
+                f"Received {val} for Metric's key name, "
+                f"which is not a valid python variable name!"
+            )
+
+        self._name = val
+
+    @property
+    def value(self):
+        """Getter for attribute `value`"""
+        return self._value
+
+    @value.setter
+    def value(self, val):
+        """Setter for attribute `value`"""
+        if self.min_value <= val and val <= self.max_value:
+            self._value = val
+            return
+
+        raise ValueError(
```

Comment:
Could you please use the raise_exception utility to ensure the exception is logged properly in a log file if a log file is configureD?